dashboard/app/models/levels/blockly.rb

```diff
@@ -607,6 +607,19 @@ def localized_function_blocks_xml(block_xml)
       )
       original_function_name = function_name.content
       function_name.content = localized_name if localized_name
+
+      # since we call the function by translated name (in `procedures_callnoreturn` blocks),
+      # we should also translate the `id` attribute that defines the function name
```

Comment:
And is calling by the translated name the right thing to do? I would expect `id` to be an attribute which is never translated. Would it be reasonable to change the calling code to use the untranslated id? I'm guessing not since it is probably generated Blockly code, but I wanted to try asking.

Reply:
Currently, we call the function by translated name but define the function by the original English name (the `id` attribute).

So, we either translate the name of the function or call it by its original English name.

I chose the first option because it doesn't impact the level UX. 

I also tested this in Hebrew, Chinese and two Arabic languages, and there were no issues with defining functions in these languages.

Reply:
@mgc1194 what do you think?

Reply:
Molly took a look and suggested to get Mike's opinion on Monday.